File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        left=0
        right=1
        maxP=0
        while right < len(prices):
            if(prices[left]<prices[right]):
                profit = prices[right] - prices[left]
                maxP=max(maxP,profit)
            else:
                left=right
            right+=1
        return maxP
        
        

File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
       int maxProfit(vector<int>& prices) {
        if(prices.size()<2)
            return 0;
        int left=0;
        int right=1;
        int res=0;
        while(right<prices.size()){
            res=max(res,prices[right]-prices[left]);
            if(prices[left]>prices[right])
                left=right;
            right++;
        }
        return res;
    }
};